Refactor/save bench to file

yumetodo opened the pull request:
<blockquote class="twitter-tweet" data-lang="ja"><p lang="ja" dir="ltr"><a href="https://twitter.com/yumetodo">@yumetodo</a> そもそもrender時benchmarkをOFFに方がいい。更新が早すぎてcrashの可能性もある</p>&mdash; 全くCloudFlareは最高だぜ (@MaverickTse) <a href="https://twitter.com/MaverickTse/status/792643345340329984">2016年10月30日</a></blockquote>


ということでファイルに書き出すpatch

yumetodo committed:
save benchmark result to file

yumetodo committed:
disable benchmark when USECLOCK  is not defined

MaverickTse commented on SigColorFastAviUtl/sigcolorfastaviutl.cpp:
```diff
@@ -506,15 +498,9 @@ BOOL func_proc_sd(FILTER *fp, FILTER_PROC_INFO *fpip) // This is the main image
-		fp->exfunc->filter_window_update(fp);
+		const auto end_sd = ch::steady_clock::now();
+		const auto elapsed_ns = std::to_string(ch::duration_cast<ch::nanoseconds>(end_sd - start_sd).count());
+		logfile << "SDeCon," + elapsed_ns << std::endl;
```
This will write to the diskdrive every 10ms or so and only writing a few byte... not good for HDD health (this is like the BT clients in early days without sufficient buffering...). Can you modify it so that:
- initially write to a buffer
- write out the buffer when it reach 4KB size or on file/program close

Also, your log file implementation need to associate with a new checkbox on the UI. 

For the current implementation, I'll try finding if any function to disable the benchmark on its own when rendering to file.

MaverickTse reviewed (changes requested):
Need to consider effect on HDD health. as this is frequent writing with little data, very much the same as BT clients in early days.

yumetodo committed:
Merge remote-tracking branch 'refs/remotes/upstream/master'

Conflicts:
	SigColorFastAviUtl/sigcolorfastaviutl.cpp

yumetodo committed:
stop io when process video

output file at exit callback function

yumetodo committed:
fix compiler warning

ref: 809bba5a7fd135d2e2e7274dd98adcf3c3292791

yumetodo committed:
stop creating benchmark file when logbuf is empty

yumetodo commented:
@MaverickTse changed to stop io when process video.

MaverickTse commented:
I said you need a NEW checkbox, not removing the title bar display!!!
That means set CHECK_N to 6 and add a new label for it.
I have already added a function to disable the benchmark checkbox when saving, so don't wipe it out.

yumetodo commented:
>not removing the title bar display!!!

!?

...I was missed 

>Also, your log file implementation need to associate with a new checkbox on the UI.

this. I'm trying to fix this.

MaverickTse commented:
take your time, no rush, don't work overnight

yumetodo committed:
fix compile warning

yumetodo committed:
(bug)Why AviUtl fail to load this plugin? What's the cause?

add checkbox and add option to print bench-info to title-bar

yumetodo commented:
![image](https://cloud.githubusercontent.com/assets/10869046/19844312/d0371f1a-9f6e-11e6-92a5-af4aa01751bb.png)
なんで``WM_COMMAND``が来てかつ``wparam``が``50344749``のときに``func_WndProc_con``がコケるんだ・・・？なんか影響のある変更したっけか

MaverickTse commented:
>なんでWM_COMMANDが来てかつwparamが50344749のときにfunc_WndProc_conがコケるんだ・・・？なんか影響のある変更したっけか

I'm also experiencing strange error. Previously, I tried adding an extra value to 
`int		check_default[] = { 1, 0, 0, 0 };`
then AviUtl crash immediately on startup :-\

Maybe `FILTER_DLL**  GetFilterTableList(void)` has bug, I'm not sure.

Anyway, try keeping check_defaut to 4 items and see if the error goes away.

yumetodo commented:
I think use config file to enable benchmark is best way. How do you think?

MaverickTse commented:
Just submitted a new patch. Try again.

yumetodo committed:
Merge remote-tracking branch 'refs/remotes/upstream/master'

Conflicts:
	SigColorFastAviUtl/sigcolorfastaviutl.cpp

yumetodo committed:
set 0(nullptr) last element of all array which is passed to AviUtl 

on WM_COMMAND event(wparam=50344749), AviUtl will crash.

yumetodo committed:
fix bug stopping benchmark when "disable benchmark when export" is off and export start

yumetodo committed:
Only the array which used as return value of GetFilterTableList must be finish with nullptr

It is not necessary seting 0(nullptr) last element of arrays which is used to init FILTER_DLL struct

yumetodo commented:
fixed!

MaverickTse commented:
お疲れ様でした　👍
